### PR TITLE
#1139 Clarify guidance in get started and programming strategy guide and reference updates

### LINKED ISSUE #1139: Community Outsourcing of documenation for devel to main PR
Thanks for helping us!!
If you choose to review any of the vignettes listed here, please edit the list and **replace "admiral-non-core-dev" with your name**

## General Check List for Each Section Below

- [ ] Free of spelling and grammar mistakes
- [ ] Links Work and directs you to intended place
- [ ] R code returns expected results
- [ ] Deprecated or soon to be deprecated functions are not used in code

## Readme and Get Started
- [x]  Readme / Readme.md  (Reviewed by:  @thomas-neitmann)
- [x]  Get Started / admiral.Rmd  (Reviewed by:  Eric Simms)

##  User Guides

- [ ]  Creating ADSL / admiral.Rmd  (Reviewed by:  Pooja)
- [x]  Creating an OCCDS ADaM / occds.Rmd  (Reviewed by:  Gordon Miller)
- [x]  Creating a BDS Finding ADaM /  bds_finding.Rmd  (Reviewed by: Nathan Kosiba)
- [x]  Creating a BDS Exposure ADAm / - bds_exposure.Rnd  (Reviewed by:  Nathan Kosiba)
- [x]  Creating a BDS Time-to-Event ADaM - bds_tte.Rmd  (Reviewed by:  Nathan Kosiba)
- [x]  Date and Time Imputation / imputation.Rmd  (Reviewed by:  Ross Farrugia)
- [x]   Queries Dataset Documentation /  queries_dataset.Rmd  (Reviewed by: Stefan)
- [x]  FAQ /  faq.Rmd   (Reviewed by:  Ross Farrugia)

## Developer Guides

- [x]  Contirubtion to admiral / contribution_model.Rmd  (Reviewed by:  Ross Farrugia)
- [x]  Development Process / development_process.Rmd  (Reviewed by:  Ross Farrugia)
- [x] Programming Strategy / programming_strategy.Rmd  (Reviewed by:  Ben Straub)
- [ ] Guidance for git and GitHub Usage / git_usage.Rmd  (Reviewed by:  Ben Straub)
- [x]  Writing Vignettes / writing_vignettes.Rmd  (Reviewed by:  @thomas-neitmann)
- [x]  Pull Request Review Guidance / pr_review_guidance.Rmd  (Reviewed by:  @thomas-neitmann)
- [x]  Unit Test Test Guidance / unit_test_guidance.Rmd  (Reviewed by:  @thomas-neitmann)

## References
- [x] Randomly Select ten Reference links  (Reviewed by:  Stefan Bundfuss)
- [ ] Randomly Select ten Reference links  (Reviewed by:  Eric Simms)

### LINKED ISSUE #1147: Bug: filter_if is falling over in Example on Website
### What happened?

Examples looks like it is falling over on website

image

### Session Information

Found on website

### Reproducible Example

Unsure on the first one example

If I change filter_if(admiral_vs, rlang::quo(VSTESTCD == "Weight")) to filter_if(admiral_vs, rlang::quo(VSTESTCD == "WEIGHT")) it produces a results.

### PR DESCRIPTION
**Pull request by bms63**
Close #1139 Close #1147

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
> Could you set `eval=TRUE`? Otherwise the result is not displayed.

The result is stated in the example text.  Maybe I should do evel=TRUE and then hide result just so it evaluated and throws error if something happens to the function

**Comment by bms63** [comment]
Hi @thomas-neitmann @bundfussr  @millerg23 Hi all. 

 Hoping you all can finish this PR for me either by reviewing or making the final updates when you sign on!
 
 42 files!  Most updates are the derive_vars_dt() and embedded example for impute_dtc that was not rendering on the website.  These are populated throughout a lot of files.  There are a few other cosmetic changes.  Please and thank you.

File: R/derive_vars_dy.R [hunk 1]
```diff
@@ -44,37 +44,43 @@
...
+#'   "2014-01-17", "2014-02-01"
 #' ) %>%
-#'  mutate(TRTSDT = ymd(TRTSDT),
-#'         DTHDT = ymd(DTHDT))
+#'   mutate(
+#'     TRTSDT = ymd(TRTSDT),
+#'     DTHDT = ymd(DTHDT)
+#'   )
 #'
 #' derive_vars_dy(datain,
```
Comment (millerg23) [inline R/derive_vars_dy.R:71]: Should `datain` on line 71 be on next line to keep function parameters aligned.

File: R/derive_vars_dy.R [hunk 2]
```diff
@@ -84,29 +90,36 @@ derive_vars_dy <- function(dataset,
...
+        ),
+        paste0(bad_vars, collapse = ", ")
+      )
     abort(err_msg)
   }
 
   dy_vars <- if_else(source_names == "",
-                     stringr::str_replace_all(vars2chr(source_vars), "(DT|DTM)$", "DY"),
-                     source_names)
+    stringr::str_replace_all(vars2chr(source_vars), "(DT|DTM)$", "DY"),
```
Comment (millerg23) [inline R/derive_vars_dy.R:104]: should `source_names` on line 93 be on next line? to be aligned with other parameters to `if_else`

File: R/derive_vars_dy.R [hunk 3]
```diff
@@ -84,29 +90,36 @@ derive_vars_dy <- function(dataset,
...
+      rename_at(
+        vars(ends_with("temp")),
+        ~dy_vars
+      )
   } else {
     dataset %>%
       mutate(!!sym(dy_vars) :=
-               compute_duration(start_date = !!reference_date, end_date = !!source_vars[[1]])
-             )
+        compute_duration(start_date = !!reference_date, end_date = !!source_vars[[1]]))
```
Comment (millerg23) [inline R/derive_vars_dy.R:123]: should `!!sym(dy_vars)` be on next line to align `mutate` parameters.

File: R/derive_vars_dy.R (hunk 1 above)
Comment (bundfussr) [reply R/derive_vars_dy.R:71]: Yes

File: R/derive_vars_dy.R (hunk 2 above)
Comment (bundfussr) [reply R/derive_vars_dy.R:104]: Yes

File: R/derive_vars_dy.R (hunk 3 above)
Comment (bundfussr) [reply R/derive_vars_dy.R:123]: Yes

**Review by millerg23 — COMMENTED** [review]
A few minor comments about function parameter alignment. Looks good!

**Comment by millerg23** [comment]
Thanks for confirming @bundfussr I can make these minor updates, and push through for final Approval.

**Comment by millerg23** [comment]
@bundfussr @thomas-neitmann can you approve to trigger Check Templates? Thanks!

File: R/derive_var_ontrtfl.R [hunk 4]
```diff
@@ -122,10 +122,10 @@
 #' )
 #'
 #' advs <- tibble::tribble(
-#'   ~USUBJID, ~ADTM,                ~TRTSDTM,                   ~TRTEDTM,                   ~TPT,
+#'   ~USUBJID, ~ADTM, ~TRTSDTM, ~TRTEDTM, ~TPT,
```
Comment (bundfussr) [inline R/derive_var_ontrtfl.R:125]: Can we avoid removing the alignment?

Comment (millerg23) [reply R/derive_var_ontrtfl.R:125]: > Can we avoid removing the alignment?

Would we need to update STYLER checks? As I think adding indentation will lead to STYLER warnings.

Comment (thomas-neitmann) [reply R/derive_var_ontrtfl.R]: Just fixed this,

**Review by thomas-neitmann — APPROVED** [review]

**Review by bundfussr — APPROVED** [review]

**Comment by bundfussr** [comment]
@bms63 , @thomas-neitmann , all checks passed.

Should we merge it or is there anything outstanding?

**Pull request merged**